flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -405,8 +405,9 @@ public void onComplete(Throwable failure, Object success) throws Throwable {
 					}
 					else {
 						if (!(success.equals(Messages.getAcknowledge()))) {
-							markFailed(new Exception("Failed to deploy the task to slot " + slot +
-									": Response was not of type Acknowledge"));
+							markFailed(new Exception("Failed to deploy the task to slot.Response was not of type Acknowledge.\n" +
+									"Please check sufficient memory is available for task threads for " + this + "\n" +
```

Comment:
I think the memory message is too specific. That may have been the point when the error was observed, but I assume that this is a rare case.

The message could simple be:
```java
new Exception("Failed to deploy the task to slot. Response was not of type 'Acknowledge', but was " + success + "\nSlot Details: " + slot);
```